src/components/fx/layout_defaults.js

```diff
@@ -9,40 +9,24 @@
 'use strict';
 
 var Lib = require('../../lib');
+var isUnifiedHover = require('./helpers').isUnifiedHover;
 var layoutAttributes = require('./layout_attributes');
+var handleHoverModeDefaults = require('./hovermode_defaults');
 
 module.exports = function supplyLayoutDefaults(layoutIn, layoutOut, fullData) {
     function coerce(attr, dflt) {
         return Lib.coerce(layoutIn, layoutOut, layoutAttributes, attr, dflt);
     }
 
-    var clickmode = coerce('clickmode');
-
-    var dragMode = coerce('dragmode');
-    if(dragMode === 'select') coerce('selectdirection');
-
-    var hovermodeDflt;
-    if(layoutOut._has('cartesian')) {
-        if(clickmode.indexOf('select') > -1) {
-            hovermodeDflt = 'closest';
-        } else {
-            // flag for 'horizontal' plots:
-            // determines the state of the mode bar 'compare' hovermode button
-            layoutOut._isHoriz = isHoriz(fullData, layoutOut);
-            hovermodeDflt = layoutOut._isHoriz ? 'y' : 'x';
-        }
-    } else hovermodeDflt = 'closest';
-
-    var hoverMode = coerce('hovermode', hovermodeDflt);
+    var hoverMode = handleHoverModeDefaults(layoutIn, layoutOut, fullData);
```

Comment:
We talked about short-circuiting this one, right? something like:
```js
var hoverMode = layoutOut.hovermode;
if (hoverMode === undefined) {
    hoverMode = handleHoverModeDefaults(layoutIn, layoutOut, fullData);
}
```
Is that a possibility? Or would that not work for some reason?

Reply:
`hovermode` and `clickmode` use special `coerce` function which returns previously coerced value.
https://github.com/plotly/plotly.js/blob/abf3a557080ae16b25ca0da57d88ba956a535b75/src/components/fx/hovermode_defaults.js#L14-L20